charts/opentelemetry-collector/templates/statefulset.yaml

```diff
@@ -17,6 +17,11 @@ spec:
   podManagementPolicy: {{ .Values.statefulset.podManagementPolicy }}
 {{- if not .Values.autoscaling.enabled }}
   replicas: {{ .Values.replicaCount }}
+{{- end }}
+{{- if and ( semverCompare ">= 1.23-0" .Capabilities.KubeVersion.Version ) ( .Values.statefulset.persistentVolumeClaimRetentionPolicy.enabled )  }}
```

Comment:
I'd prefer not to add complexity in the template to check k8s version. A user on a version of Kubernetes that doesn't support `persistentVolumeClaimRetentionPolicy` should not use this feature.  Can you add a comment in the values.yaml that specifies the first k8s version the feature is supported in?

Reply:
Ok that makes sense.

I have removed the semverCompare and added a comment in values.yaml now.